brage-import/src/main/java/no/sikt/nva/brage/migration/merger/S3MultipartCopier.java

```diff
@@ -121,14 +125,17 @@ private void performSimpleCopy(S3Client s3Client) {
         s3Client.copyObject(copyObjRequest);
     }
 
-    private void performMultiPartCopy(S3Client s3Client, HeadObjectResponse headOfObjectToCopy) {
+    private void performMultiPartCopy(S3Client s3Client, Context context, HeadObjectResponse headOfObjectToCopy) {
         var request = initiateMultiUploadRequest(headOfObjectToCopy);
         var response = s3Client.createMultipartUpload(request);
         try {
             long position = 0;
             int partNumber = 1;
             long totalSize = headOfObjectToCopy.contentLength();
             while (position < totalSize) {
+                if (context.getRemainingTimeInMillis() < TIMEOUT_THRESHOLD_30_SECONDS) {
```

Comment:
what's happening here, when getRemainingTimeInMillis() is less then 30 000ms, then throw timeout

Reply:
Yes, if there is less then 30 seconds left to proceed the record, we throw exception and persist error report. Before exception was thrown without persisting report, which made it difficult to find missing record.

Reply:
ok :-) sounds good then 👍